src/main/java/seedu/address/storage/JsonAdaptedRole.java

```diff
@@ -0,0 +1,128 @@
+        deadline = source.getDeadline().value.toString();
+        description = source.getDescription().value;
+        stipend = source.getStipend().value;
+    }
+
+    //    @JsonValue
+    //    public String getRoleName() {
+    //        return name;
+    //    }
+    //
+    //    @JsonValue
+    //    public String getRoleStatus() {
+    //        return status;
+    //    }
+    //
+    //    @JsonValue
+    //    public String getRoleDeadline() {
+    //        return deadline;
+    //    }
+    //
+    //    @JsonValue
+    //    public String getRoleDescription() {
+    //        return description;
+    //    }
+    //
+    //    @JsonValue
+    //    public String getRoleStipend() {
+    //        return stipend;
+    //    }
+
+    /**
+     * Converts this Jackson-friendly adapted tag object into the model's {@code Role} object.
+     *
+     * @throws IllegalValueException if there were any data constraints violated in the adapted role.
+     */
+    public Role toModelType() throws IllegalValueException {
+        if (name == null) {
+            throw new IllegalValueException(String.format(MISSING_FIELD_MESSAGE_FORMAT,
+                    RoleName.class.getSimpleName()));
+        }
+        if (!RoleName.isValidName(name)) {
+            throw new IllegalValueException(RoleName.MESSAGE_CONSTRAINTS);
+        }
+        final RoleName modelRoleName = new RoleName(name);
+
+        if (status == null) {
+            throw new IllegalValueException(String.format(MISSING_FIELD_MESSAGE_FORMAT,
+                    Status.class.getSimpleName()));
+        }
+        if (!Status.isValidStatus(status)) {
+            throw new IllegalValueException(Status.MESSAGE_CONSTRAINTS);
+        }
+        final Status roleStatus = new Status(status);
+
+        if (deadline == null) {
+            throw new IllegalValueException(String.format(MISSING_FIELD_MESSAGE_FORMAT,
+                    Deadline.class.getSimpleName()));
+        }
+        if (!Deadline.isValidDeadline(deadline)) {
+            throw new IllegalValueException(Deadline.MESSAGE_CONSTRAINTS);
+        }
+        final Deadline roleDeadline = new Deadline(deadline);
+
+        if (description == null) {
+            throw new IllegalValueException(String.format(MISSING_FIELD_MESSAGE_FORMAT,
+                    Description.class.getSimpleName()));
+        }
+        if (!Description.isValidDescription(description)) {
+            throw new IllegalValueException(Description.MESSAGE_CONSTRAINTS);
+        }
+        final Description roleDescription = new Description(description);
+
+        if (stipend == null) {
+            throw new IllegalValueException(String.format(MISSING_FIELD_MESSAGE_FORMAT,
+                    Stipend.class.getSimpleName()));
+        }
+        if (!Stipend.isValidStipend(stipend)) {
+            throw new IllegalValueException(Stipend.MESSAGE_CONSTRAINTS);
+        }
+        final Stipend roleStipend = new Stipend(stipend);
```

Comment:
So if any of the fields are empty, it will store an empty string?

Reply:
If the fields are empty it will throw an exception. So the Role object won't be created